pkg/transforms/batch.go

```diff
@@ -117,7 +116,9 @@ func (batch *BatchConfig) Batch(edgexcontext *appcontext.Context, params ...inte
 			if len(batch.batchData) < batch.batchThreshold {
 				return false, nil
 			}
-			batch.done <- true
+			if batch.done != nil {
```

Comment:
Why would the channel not be created? Was if getting stuck here?

Reply:
Dug in more and think it get it. Channel not needed when bacth by count. If that is correct , please add a `why` comment explaining the nil check.

Reply:
no listeners so it would hang